ios/Classes/components/card/session/CardSessionComponent.swift

```diff
@@ -69,14 +71,15 @@ class CardSessionComponent: BaseCardComponent {
     }
 
     private func setupFinalizeComponentCallback() {
-        (sessionHolder.sessionDelegate as? CardSessionFlowDelegate)?.finalizeAndDismissHandler = finalizeAndDismissSessionComponent
-        (sessionHolder.sessionDelegate as? CardSessionFlowDelegate)?.componentId = componentId
+        (sessionHolder.sessionDelegate as? ComponentSessionFlowDelegate)?.finalizeAndDismissHandler = finalizeAndDismissSessionComponent
+        (sessionHolder.sessionDelegate as? ComponentSessionFlowDelegate)?.componentId = componentId
     }
 
     func finalizeAndDismissSessionComponent(success: Bool, completion: @escaping (() -> Void)) {
         finalizeAndDismiss(success: success, completion: { [weak self] in
-            self?.sessionHolder.reset()
             completion()
+            self?.sessionHolder.reset()
```

Comment:
As a rule of thumb. If you need to use a weak self more than once, please use `guard let self else { return }` to exit early and prevent uses of optional self